src/DssDirectDepositAaveDai.sol

```diff
@@ -236,6 +249,45 @@ contract DssDirectDepositAaveDai {
         return _rdiv(_add(stableDebt.totalSupply(), variableDebt.totalSupply()), targetUtil);
     }
 
+    // Use this to determine whether exec() should be called based on your interest rate deviation threshold
+    // This assumes normal operation, culled / global shutdown should be handled externally
+    // Also assumes no liquidity issues
+    function shouldExec(uint256 interestRateTolerance) external view returns (bool) {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+
+        (uint256 daiDebt,) = vat.urns(ilk, address(this));
+        uint256 _bar = bar;
+        if (_bar == 0) {
+            return daiDebt > 0;     // Always attempt to close out if we have debt remaining
+        }
+
+        (,, uint256 currVarBorrow) = interestStrategy.calculateInterestRates(
+            address(adai),
+            dai.balanceOf(address(adai)),
+            stableDebt.totalSupply(),
+            variableDebt.totalSupply(),
+            0,
+            0
+        );
+
+        uint256 deviation = _rdiv(currVarBorrow, _bar);
+        if (deviation < RAY) {
+            // Unwind case
+            return daiDebt > 0 && (RAY - deviation) > interestRateTolerance;
+        } else if (deviation > RAY) {
+            // Wind case
+            (uint256 Art,,, uint256 line,) = vat.ilks(ilk);
+            return _sub(line, Art*RAY) > RAD && (deviation - RAY) > interestRateTolerance;  // Allow for rounding errors, DC = full if 1 Dai away
+        } else {
+            // No change
+            return false;
+        }
+        
+        return deviation - RAY > interestRateTolerance;
```

Comment:
how do we hit this case? I seem like the `if`s above will handle all of the branches?

Reply:
Just realized you also mentioned it. Yeah this needs to be removed.